tests/phpunit/tests/reusable-blocks/test-blocks.php

```diff
@@ -105,10 +105,96 @@ public function it_should_get_ids_from_post() {
 			'args'   => [ $post->post_content ],
 			'return' => $blocks,
 		]);
+		
+		// Second stack
+		$block = $this->getMockBuilder( 'WP_Post' )->disableOriginalConstructor()->getMock();
+		$block->post_content = 'some content';
+
+		\WP_Mock::userFunction( 'get_post', [
+			'args'   => [ $block_id ],
+			'return' => $block,
+		]);
+		\WP_Mock::userFunction( 'parse_blocks', [
+			'args'   => [ $block->post_content ],
+			'return' => [],
+		]);
+		
 		$subject = new Blocks();
 		$this->assertEquals(
 			[ $block_id ],
-			$subject->getIdsFromPost( $post_id )
+			$subject->getChildrenIdsFromPost( $post_id )
+		);
+		unset( $GLOBALS['wp_version'] );
+	}
+
+
+
+	/**
+	 * @test
+	 * @group wpmlcore-6598
+	 */
+	public function it_should_get_nested_block_ids_from_post() {
+		$GLOBALS['wp_version'] = '5.1.0';
+
+		$post_id        = 123;
+		$block_id       = 456;
+		$child_block_id = 789;
+
+		// First stack
```

Comment:
@strategio I have no idea what `First stack`, `Second stack` and `Last stack` mean. Maybe you need to move the code for each part into a member function with a more meaningful name

Reply:
Done @brucepearson, I refactored the expectations into `set_block_in_post`, easier to read now.